### PR TITLE
[Civl] Cleaned up the ticket sample

### PR DESCRIPTION
**Pull request by shazqadeer**

### CODE REVIEW

File: Test/civl/samples/ticket.bpl [hunk 1]
```diff
@@ -10,8 +10,8 @@ type {:linear "tid"} X;
 const nil: X;
```
Comment (bkragl) [inline Test/civl/samples/ticket.bpl:10]: I think it would be nice to replace this with `Option<_>` from the library. (And rename `X` to `Tid`.)

**Review by bkragl — COMMENTED** [review]
So one change here is the removal of the top level main procedure that calls `Customer` in a loop, and the necessary linear allocation code. Wondering why that isn't something useful to illustrate?

It seems there are other orthogonal improvements, like introducing the `tid` parameters, but I don't get the high-level picture. What is it?

**Comment by shazqadeer** [comment]
Constantin suggested using the ticket example to be the primary illustrator of all Civl machinery except permissions in the upcoming POPL 2024 tutorial. He made slides explaining the example. I was trying to change the code to become closer to the exposition. In particular, I dropped all the allocation of thread identifiers from the sample because permissions (specifically redistribution) will be explained in a more comprehensive using other examples (to be determined) and there is plenty going on in this example already.

Adding main back might still be useful.

Suggestions about the tutorial section on permissions are welcome.

File: Test/civl/samples/ticket.bpl (hunk 1 above)
Comment (shazqadeer) [reply Test/civl/samples/ticket.bpl:10]: Good idea. Will do.

**Comment by shazqadeer** [comment]
@bkragl : BTW, I stopped writing PR summaries since I didn't think anybody was paying attention :-). But every once in a while you pop in with some insightful question or comment. Do you actually look at the Civl PRs regularly?

**Comment by bkragl** [comment]
I do go over my github notifications, but often this happens in bursts. I try and see if I have the necessary context to contribute something, which unfortunately does not seem that often these days. I guess it's more likely for examples than it is for changes in the code. I understand that this is not a big motivation to write elaborate PR descriptions.

**Comment by shazqadeer** [comment]
Here is a PR: https://github.com/boogie-org/boogie/pull/828

**Pull request merged**